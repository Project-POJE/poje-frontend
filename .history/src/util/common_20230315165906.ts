export function timeFormat(day: string) {
  let [date, time] = day.split('T');
  time = time.split(':').slice(0, 2).join(':');
  return [date, ' ', time];
}