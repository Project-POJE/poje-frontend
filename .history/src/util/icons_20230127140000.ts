type Icons = {
  front: string[];
};

const icons = {};
export {};
