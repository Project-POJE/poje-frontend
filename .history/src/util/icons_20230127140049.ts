type Icons = {
  frontend: string[];
  backend: string[];
};

const icons: Icons = {};
export {};
