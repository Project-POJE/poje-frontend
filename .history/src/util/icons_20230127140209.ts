type Icons = {
  frontend: string[];
  backend: string[];
};

const icons: Icons = {
  frontend: [],
  backend: [],
};
export {};
