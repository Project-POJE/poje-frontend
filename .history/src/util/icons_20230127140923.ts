type Icons = {
  frontend: string[];
  backend: string[];
};

const icons: Icons = {
  frontend: [],
  backend: [],
};

const iconFolderNames = ['frontend', 'backend'];
const iconPath = '/public/public_assets/skill_icon';
iconFolderNames.forEach((folderName) => {
  const context = require.context(`${iconPath}/${folderName}`, false);
  //context.keys().map((i) => icons[].push(test(i)))
});
export {};
