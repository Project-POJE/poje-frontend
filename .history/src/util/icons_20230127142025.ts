type IconsType = {
  frontend: string[];
  backend: string[];
};

type IconsKeyType = keyof IconsType;
const a: IconsKeyType = ['fon'];
const icons: IconsType = {
  frontend: [],
  backend: [],
};

const iconFolderNames = ['frontend', 'backend'];
const iconPath = '/public/public_assets/skill_icon';
iconFolderNames.forEach((folderName) => {
  const context = require.context(`${iconPath}/${folderName}`, false);
  //context.keys().map((i) => icons[].push(test(i)))
});
export {};
