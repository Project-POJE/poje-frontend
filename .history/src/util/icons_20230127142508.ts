type IconsType = {
  frontend: string[];
  backend: string[];
};

type IconsKeyType = keyof IconsType;

const icons: IconsType = {
  frontend: [],
  backend: [],
};

const iconFolderNames: IconsKeyType[] = ['frontend', 'backend'];
const iconPath = '/public/public_assets/skill_icon';
iconFolderNames.forEach((folderName) => {
  const context = require.context(`${iconPath}/${folderName}`, false);
  context.keys().map((i) => icons[folderName].push(context(i) as string));
});
export default icons;
