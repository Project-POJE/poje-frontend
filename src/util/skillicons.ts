import { SkillItemType } from '@src/service/types/portfolio';

export type SkillIconSetType = {
  name: string;
  path: string;
};

export type SkillIconType = 'frontend' | 'backend';

export type IconListType = Record<SkillIconType, SkillItemType[]>;

export type IconListKeyType = keyof IconListType;

type SkillTypeAndNameType = {
  type: IconListKeyType;
  name: string;
  path: string;
};

export function extractSkillIconFromFolder() {
  //skill_icon 하위 폴더들의 이미지들을 가져옴
  const imgContext = require.context(
    '/public/public_assets/skill_icon',
    true,
    /.png$|.ipg$/
  );

  const icons: IconListType = {
    frontend: [],
    backend: [],
  };

  //./backend/node.png 이런애들이 Keys()에 있음
  imgContext.keys().forEach((k) => {
    const extractedSkill = extractSkillTypeAndName(k);
    console.log(k);
    if (extractedSkill) {
      const name = extractedSkill.name;
      const path = extractedSkill.path;
      icons[extractedSkill.type].push({ name, path });
    }
  });
  console.log(icons);

  return icons;
}

// 동적으로 키에 배열을 작성하고 난 후 그 키를 동적으로 가져오는 것은 불가능
function extractSkillTypeAndName(path: string): SkillTypeAndNameType | null {
  // 직업식별자에 포함된 이미지의 식별자를 가져오기
  let skillType, skillName, skillPath;
  let result = path.replace('./', ''); //추출된 스킬 front/css.png;
  const splitPathList = result.split('/');

  if (splitPathList.length === 1) return null;
  skillType = splitPathList[0] as IconListKeyType;
  skillName = splitPathList[1].split('/')[0].split('.')[0];
  skillPath = splitPathList[1].split('/')[0];
  return { type: skillType, name: skillName, path: skillPath };
}
