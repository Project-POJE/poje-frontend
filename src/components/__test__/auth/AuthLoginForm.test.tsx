import AuthLoginForm from '@src/components/auth/AuthLoginForm';
import useAuth from '@src/hooks/auth/useAuth';
import { withAllContext, withMemoryRouter } from '@src/test_util/util';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Route } from 'react-router-dom';

jest.mock('@src/hooks/auth/useAuth');
const mockedUseAuth: jest.Mock = useAuth as jest.MockedFunction<any>;

describe('AuthLoginForm', () => {
  beforeEach(() => {
    mockedUseAuth.mockImplementation(() => ({ login: jest.fn() }));
  });

  afterEach(() => {
    mockedUseAuth.mockReset();
  });

  it('처음 렌더링 화면', () => {
    const { asFragment } = render(
      withAllContext(
        withMemoryRouter(<Route path='/' element={<AuthLoginForm />} />, {
          initialEntries: ['/'],
        })
      )
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('아이디, 비밀번호 입력 후 submit 전송', async () => {
    const mockLogin = jest.fn();
    mockedUseAuth.mockImplementation(() => ({
      login: mockLogin,
    }));

    render(
      withAllContext(
        withMemoryRouter(<Route path='/' element={<AuthLoginForm />} />, {
          initialEntries: ['/'],
        })
      )
    );

    const loginInput = screen.getByPlaceholderText(
      'LoginId'
    ) as HTMLInputElement;
    const pwInput = screen.getByPlaceholderText('password') as HTMLInputElement;
    const submitBtn = screen.getByText('로그인');

    userEvent.type(loginInput, 'seungyn');
    userEvent.type(pwInput, 'seungyn');

    userEvent.click(submitBtn);

    await waitFor(() =>
      expect(mockLogin).toHaveBeenCalledWith({
        loginId: 'seungyn',
        password: 'seungyn',
      })
    );
  });
});
