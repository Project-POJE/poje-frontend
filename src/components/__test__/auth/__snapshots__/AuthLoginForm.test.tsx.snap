// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthLoginForm 처음 렌더링 화면 1`] = `
<DocumentFragment>
  <section
    class="sc-eDvSVe eZUHhS"
  >
    <div
      class="sc-jSUZER jEHtLy"
    >
      <button
        class="sc-bcXHqe fYfQRa"
      >
        <svg
          class="icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
          />
        </svg>
      </button>
      <div
        class="sc-kDvujY dOzmYf"
      >
        <span>
          계정이 없으신가요?
        </span>
        <button
          class="sc-gswNZR sc-gKPRtg fGpxWi dpcrmR"
        >
          <a
            href="/auth/signup"
          >
            SignUp
          </a>
        </button>
      </div>
    </div>
    <header
      class="sc-iBYQkv cVxthA"
    >
      <h1>
        WelCome Back
      </h1>
      <p>
        Login your account
      </p>
    </header>
    <form
      class="sc-ftTHYK iNVSoq"
    >
      <label
        class="sc-pyfCe bA-Deeb"
        for="loginId"
      >
        <span>
          LoginId
        </span>
        <input
          id="loginId"
          name="loginId"
          placeholder="LoginId"
          type="text"
        />
      </label>
      <label
        class="sc-pyfCe bA-Deeb"
        for="password"
      >
        <span>
          Password
        </span>
        <input
          id="password"
          name="password"
          placeholder="password"
          type="password"
        />
      </label>
      <button
        class="sc-gswNZR sc-jrcTuL fGpxWi cYLYkE"
      >
        로그인
      </button>
    </form>
    <button
      class="sc-ipEyDJ kKeVQ"
    >
      비밀번호를 잊어버렸습니까?
    </button>
  </section>
</DocumentFragment>
`;
